hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ec/reconstruction/ECReconstructionSupervisor.java

```diff
@@ -38,6 +39,11 @@ public class ECReconstructionSupervisor implements Closeable {
   private final StateContext context;
   private final ExecutorService executor;
   private final ECReconstructionCoordinator reconstructionCoordinator;
+  /**
+   * how many coordinator tasks currently being running.
+   */
+  private final ConcurrentHashMap.KeySetView<Object, Boolean>
```

Comment:
These seems to be copied from the ReplicationSupervisor - do we have a plan to use this for more than just counting tasks? Could we use an atomicLong as a simple counter instead?

Reply:
Yes, Just used this from ReplicationSupervisor. I thought is atomicLong, but looks like we are trying to avoid duplicate task for same command by any chance.

While adding task:
if (inProgressReconstrucionCoordinatorCounter
        .add(taskInfo.getContainerID())) {
        
 I thought this may be useful check for safety and may avoid any accidental corruption due to fight two containers. Ideally that should be protected as we would create containers, the other one should fail.